plugins/jobtype/src/azkaban/jobtype/HadoopSecureWrapperUtils.java

```diff
@@ -52,10 +53,14 @@
    *         the logged in user's tokens
    * @throws IOException
    */
-  public static UserGroupInformation createSecurityEnabledProxyUser(String userToProxy, Logger log)
-          throws IOException {
-
-    String filelocation = System.getenv(HADOOP_TOKEN_FILE_LOCATION);
+  private static UserGroupInformation createSecurityEnabledProxyUser(String userToProxy,String tokenFile, Logger log
+          ) throws IOException {
+
+    String filelocation = null;
+    if (tokenFile != null)
+      filelocation = tokenFile;
```

Comment:
This seems to be the original logic.
I've only added the ability to specify a token file from the outside